### DISCUSSION TITLE
Why do I get giant css files (>100mb) when generating my website

### DISCUSSION THREAD

**Discussion by harshcurious** [adityatelange/hugo-PaperMod#325] [Q&A]
I made my website over [here](https://github.com/harshcurious/harshcurious.github.io). But when I run the `hugo --minify` command I get giant css files (over 100 mb) and I can't even upload it to github. Is there a way for me to learn why that is happenning and how to avoid it! 

BTW: Thanks a lot for the theme and the active support you are providing. :thumbsup:

![file size](https://i.imgur.com/bWCnLAN.png)

**Comment by harshcurious** [comment]
I was trying out a few things to solve this. I managed to solve (for now) it by 
1. Updating the theme folder by running `git submodule update --init --recursive` (I had added the theme in the same format)
2. Deleting the contents of `assets/css` folder.
3. Running the `hugo --minify` command.

I still don't understand what led to the situation:question: But I hope this helps someone else :yellow_heart:

**Reply by adityatelange** [reply]
@harshcurious Frankly, I have no idea why this would happen. Maybe update hugo ? I use release 0.81(linux/amd64) and never experienced such behaviour

**Reply by harshcurious** [reply]
I had been using the extended version of Hugo 0.81 (recently updated to 0.82) on Windows 10. The only unusual thing I do it instead of publishing to the `public` directory I publish to `.\` directory.

**Reply by harshcurious** [reply]
Some more weird things happening: look at these two images
![After two builds](https://i.imgur.com/qYFc0PW.png)
This is after **two** builds with `hugo --minify`

![After three builds](https://i.imgur.com/slfzmaO.png)
This is after **three** builds.

The file size is increasing by twice every time. Also if you look at the file sizes in the earlier images, it's the same.

**Reply by adityatelange** [reply]
@harshcurious How about using `hugo --gc --minify --cleanDestinationDir` so that it doesn't use previous assets ?

**Reply by harshcurious** [reply]
Sorry but `--cleanDestinationDir` will delete my content files as well. I think it is probably because of that as well. I guess, I just have to add the additional step of clearing that folder  :pensive:.

**Reply by adityatelange** [reply]
Umm... no it won't. Read here https://gohugo.io/commands/hugo/ it is  meant to clean `public` folder before generating web-pages.

**Reply by danielfdickinson** [reply]
This sounds like a Hugo or minify library issue rather than theme issue. I would use the [Hugo forum](https://discourse.gohugo.io/) to discuss it. There was a file size ballooning issue with javacript minification in Hugo 0.84.2 at least (which the author of minify fixed).
